unstructured_ingest/v2/processes/connectors/delta_table.py

```diff
@@ -94,21 +94,34 @@ class DeltaTableUploader(Uploader):
     connection_config: DeltaTableConnectionConfig
     connector_type: str = CONNECTOR_TYPE
 
-    @requires_dependencies(["s3fs", "fsspec"], extras="s3")
+    @requires_dependencies(["boto3"], extras="delta-table")
     def precheck(self):
         secrets = self.connection_config.access_config.get_secret_value()
         if (
             self.connection_config.aws_region
             and secrets.aws_access_key_id
             and secrets.aws_secret_access_key
         ):
-            from fsspec import get_filesystem_class
+            from urllib.parse import urlparse
+
+            from boto3 import client
 
             try:
```

Comment:
We should isolate only the network call in the try-except, that way if urlparse breaks, that should be raised since it doesn't mean the connection failed. Or if this is something that could be caused by user input, wrap it in a different try-except and provide a useful message in the error that the table_uri was malformed.

Reply:
urlparse is now excluded from try-except. I believe it can't break, because of the regex on this field in the UI that doesn't allow for any error with parsing